src/CMakeLists.txt

```diff
@@ -460,6 +461,10 @@ else()
     NAMELINK_SKIP)
 endif()
 
+if(WIN32)
+  set(win32_common ceph-common)
```

Comment:
i'd avoid this.

Reply:
That was @wjwithagen suggestion [1] and IMO it was a really good one. The idea is that as opposed to Linux shared libaries, Windows DLLs can't have unresolved symbols. It was much easier to do this rather than add checks wherever ceph-common was needed.

[1] https://github.com/ceph/ceph/pull/31981#discussion_r353360732

Reply:
> 
> 
> That was @wjwithagen suggestion [1] and IMO it was a really good one. The idea is that as opposed to Linux shared libaries, Windows DLLs can't have unresolved symbols and libosd is in fact using ceph-common functions. It was much easier to do this rather than add checks wherever ceph-common was needed.
> 
> [1] [#31981 (comment)](https://github.com/ceph/ceph/pull/31981#discussion_r353360732)

@petrutlucian94 
I'm sorry about the mix of opinions. @tchaikov has to merge the PR in the end and he has final saying in this. I can only try to keep you going as much as I can.
But then I also have some discussion about my PRs over time.
And like we discussed, I really do not like the continous streams of `if(WIN32)` on a lot of places to add you windows lib.
IMHO we need something that is the CMake target `ceph-common`, which includes everything that is needed to link the common libraries. 
And perhaps it needs a different nameing..... Something like `ceph-common-extra`.
But that I'll leave up to @tchaikov to discuss with you

Reply:
Luckily we don't need libosd, so we can defer this discussion to the follow up PR. Renaming win32_common to something else sounds good as well.

Reply:
@wjwithagen `if(WIN32)` is indeed a curse. i think we can use generator expression in cmake to introduce platform specific linkage of flags, something like
```cmake
target_link_libraries(
  some_library
  PUBLIC
    some_other_library
    $<$<PLATFORM_ID:Windows>:wsock32>
  PRIVATE
    $<$<PLATFORM_ID:Linux>:some_linux_library>
)
```

Reply:
@tchaikov 
It already looks a lot nicer.
But then still we are adding this set of GenExpr to every executable build that needs these extra libraries. So I guess it sort of depends on how many places it should.
And from memory, it was quite a few, since it triggered me to make a remark about it.

Having said all this, I read that for this PR it is no longer needed, so we should take this up, once @petrutlucian94 runs into it again.
And then deal with it.

Reply:
> @wjwithagen `if(WIN32)` is indeed a curse. i think we can use generator expression in cmake to introduce platform specific linkage of flags, something like
> 
> ```cmake
> target_link_libraries(
>   some_library
>   PUBLIC
>     some_other_library
>     $<$<PLATFORM_ID:Windows>:wsock32>
>   PRIVATE
>     $<$<PLATFORM_ID:Linux>:some_linux_library>
> )
> ```

That seems fine as well, thanks for the suggestion.

Reply:
@tchaikov I've updated the subsequent PR to use this syntax: https://github.com/ceph/ceph/pull/32779/commits/429fc2e0033dc527518145c775450a900218c77a#diff-edee2648008d3d163ca405b0c44c1cedR28

For what is worth, there were 16 occurrences.